### PR TITLE
tests: protoassert in pg tests

### PR DESCRIPTION
**Pull request by janisz**
- #11531
- #11533
- #11503
- #11507
- #11575
- #11571
- #11572  
- https://github.com/stackrox/stackrox/pull/11582
- https://github.com/stackrox/stackrox/pull/11583 
- https://github.com/stackrox/stackrox/pull/11585
- https://github.com/stackrox/stackrox/pull/11588 :point_left:
- https://github.com/stackrox/stackrox/pull/11618 
- https://github.com/stackrox/stackrox/pull/11619

### CODE REVIEW

**Review by rhybrillou — APPROVED** [review]
The comments can be addressed as follow-ups

File: Makefile [hunk 1]
```diff
@@ -777,6 +777,9 @@ roxvet: skip-dirs := operator/pkg/clientset \
 roxvet: $(ROXVET_BIN)
 	@echo "+ $@"
 	@# TODO(ROX-7574): Add options to ignore specific files or paths in roxvet
+	$(SILENT)go list -e ./... \
+	    | $(foreach d,$(skip-dirs),grep -v '$(d)' |) \
+	    xargs -n 1000 go vet -vettool "$(ROXVET_BIN)" -donotcompareproto -tags "sql_integration"
```
[the remark this answers is not in the document]
Comment (janisz) [reply Makefile:782]: https://issues.redhat.com/browse/ROX-24970

File: central/complianceoperator/v2/integration/datastore/datastore_impl_test.go [hunk 2]
```diff
@@ -345,8 +345,8 @@ func (s *complianceIntegrationDataStoreTestSuite) TestRemoveComplianceIntegratio
 	s.NoError(err)
 	s.Equal(len(ids)-1, len(integrations))
 	s.NotContains(integrations, testIntegrations[0])
```
[the remark this answers is not in the document]
Comment (janisz) [reply central/complianceoperator/v2/integration/datastore/datastore_impl_test.go:347]: Good catch, I didn't know that there is such function. I'll add it

File: central/complianceoperator/v2/rules/datastore/datastore_impl_test.go [hunk 3]
```diff
@@ -380,7 +381,7 @@ func (s *complianceRuleDataStoreTestSuite) TestGetRulesByCluster() {
 		retrievedObjects, err := s.dataStore.GetRulesByCluster(tc.testContext, tc.clusterID)
 		s.Require().NoError(err)
 		s.Require().Equal(tc.expectedCount, len(retrievedObjects))
```
[the remark this answers is not in the document]
Comment (janisz) [reply central/complianceoperator/v2/rules/datastore/datastore_impl_test.go:383]: actually we can remove this line as we check for slices equality line below

File: central/complianceoperator/v2/integration/datastore/datastore_impl_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (janisz) [reply central/complianceoperator/v2/integration/datastore/datastore_impl_test.go:347]: https://github.com/stackrox/stackrox/pull/11713

**Pull request merged**